data_fetcher.py

```diff
@@ -0,0 +1,110 @@
+import os
+import shutil
+import json
+import pandas as pd
+import yfinance as yf
+from datetime import datetime
+
+class DataFetcher:
+    """
+    This module downloads data from yahoo finance and saves it locally in a downloads directory
+    """
+    def __init__(self, directory_path: str, import_start_date: datetime=datetime(2008,1,1)) -> None:
+        self.module_version = 1
+        self.directory_path = directory_path
+        self.start_date = import_start_date
+        
+    def init_downloads_directory(self, clear_contents: bool=False) -> None:
+        """
+        checks if a downloads directory exists, and if not, creates it
+        if clear flag is set to true, directory is cleared
+        """
+        # if not os.path.exists(self.directory_path):
+        #     os.makedirs(self.directory_path)
+        #     print("Directory created:", self.directory_path)
+        # else:
+        #     print("Directory already exists:", self.directory_path)
+
+        if os.path.exists(self.directory_path):
+            if clear_contents:
+                shutil.rmtree(self.directory_path)
```

Comment:
I think we should create a backup first and then remove all date.

Reply:
managing backups can generate some mess of a logic
all kinds of cases...need to check if the target backup directory already exists and so on

since this option is to be set deliberately by the user, I rather leave it to his responsibility
if we see that is required we can add that logic later